docs/storefront/headless/routes.mdx

```diff
@@ -0,0 +1,220 @@
+
+The `route` node supports 301 Redirects that you have added to the store. If the URL provided as input to the `route` references a redirect, then the response will return a `Redirect` node which comes in two varieties: a manual redirect or a dynamic redirect.
+
+A `ManualRedirect` returns a fixed destination URL. You will need a new `route` query to attempt to pull the content at that destination URL. If the destination URL itself is a redirect, this can create a chain of redirects requiring multiple queries.
+
+A "dynamic redirect" returns an entity type and entity ID of an existing store resource. There are five types of dynamic redirects: `ProductRedirect`, `BrandRedirect`, `BlogPostRedirect`, `CategoryRedirect`, and `PageRedirect`. Once the entity type and ID are retrieved, you can initiate a new query to pull that content from the appropriate resource node. With dynamic redirects, you can retrieve the target entity directly within the `route` query response, avoiding an additional request. This is detailed below.
+
+## Redirect behavior of `route` node
+
+It is possible to supply a `redirectBehavior` input argument to the `route` node which supports two values:
+
+* **IGNORE**(default): 
+
+   * The API ignores redirects and prioritizes direct entity lookups via custom URLs. 
+   * If both a redirect and an entity URL exist for the same path, both the `redirect` and `node` nodes return non-null values. 
+   * This behavior ensures backward compatibility.
+
+* **FOLLOW**: 
+
+  * The `node` node returns the target entity (e.g., category, product, blog, etc.) if the redirect is dynamic. 
+  * For static/manual redirects, the `node` node returns `null` (since there’s no associated entity), while the `redirect` node provides the redirect details.
+
+## Example queries
+
+### Query: Fetching route details
+
+This example retrieves details about an object (e.g., Product, Brand, or Category) based on a given URL path.
+
+<Tabs items={['Request', 'Response']}>
+  <Tab>
+    ```graphql filename="Query route details" showLineNumbers copy
+
+     query LookUpUrl(
+       $urlPath: String!
+       # Use GraphQL Query Variables to provide a path
+     ) {
+       site {
+         route(path: $urlPath) {
+           node {
+             __typename
+             id
+             # A different response is returned based on which type of object was matched
+             ... on Category {
+               name
+               description
+             }
+             ... on Brand {
+               name
+               defaultImage {
+                 url(width: 200)
+               }
+             }
+             ... on Product {
+               name
+               description
+               images {
+                 edges {
+                   node {
+                     url(width: 500, height: 500)
+                   }
+                 }
+               }
+             }
+             ... on Redirect {
+               toUrl
+             }
```

Comment:
I think this is in the wrong spot. It looks like it is under the `node` block which it does not belong as it is not a type of object returned by the `node`. It should exist under a `redirect` node that sites at the same level as the `node` node. Or it could be removed altogether.

If you do place it under the `redirect` node then be sure to add it to the output as it will be `redirect: null` in this case.